RadixWallet/Clients/CloudBackupClient/CloudBackupClient+Live.swift

```diff
@@ -142,20 +143,55 @@ extension CloudBackupClient {
 			try? userDefaults.setLastCloudBackup(result, of: profile)
 		}
 
+		@Sendable
+		func performAutomaticBackup(_ profile: Profile, timeToCheckIfClaimed: Bool) async {
+			let needsBackUp = profile.appPreferences.security.isCloudProfileSyncEnabled && profile.header.isNonEmpty
+			let lastBackup = userDefaults.getLastCloudBackups[profile.id]
+			let lastBackupSucceeded = lastBackup?.result == .success && lastBackup?.saveHash == profile.saveHash
+
+			let shouldBackUp = needsBackUp && !lastBackupSucceeded
+			let shouldCheckClaim = shouldBackUp || timeToCheckIfClaimed
+
+			guard shouldBackUp || shouldCheckClaim else { return }
+
+			let existingRecord = try? await fetchProfileRecord(.init(recordName: profile.id.uuidString))
+
+			let backedUpID = { try? existingRecord.map(getProfileHeader)?.lastUsedOnDevice.id }
+
+			let didTransferBackProfile: Bool
+			if shouldCheckClaim, let id = backedUpID(), await !profileStore.isThisDevice(deviceID: id) {
```

Comment:
Can `backedUpID` be ever `nil`? or it is just semantics

Reply:
`existingRecord` is the backed up profile, and it can definitely be `nil`. In theory `getProfileHeader` could also throw, if the CKRecord doesn't contain all the properties.

Reply:
ok, so in this case should go forward and upload the current profile as to repair the record in the cloud?

Reply:
Yes, that's what the `didTransferBackProfile` flag is for

Reply:
...now renamed to `shouldReclaim`